### PR TITLE
chore: REV-1493: Results of 2021-05-26 make upgrade

### PR DESCRIPTION
**Pull request by bmedx**
## Description

Periodic chore to upgrade non-pinned Python packages to the latest versions. The diffs are messy due to the new format used by pip tools. Easier to read diffs for the .txt files are included in PR comments.

Django changes are bug fixes and security patches.

### CODE REVIEW

File: ecommerce/core/migrations/0065_ecommercefeatureroleassignment_applies_to_all_contexts.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.23 on 2021-05-26 20:05
```
Comment (bmedx) [inline ecommerce/core/migrations/0065_ecommercefeatureroleassignment_applies_to_all_contexts.py:1]: This model only has 5 rows in it, so this should be a fast migration.

File: ecommerce/extensions/refund/migrations/0008_auto_20210526_2005.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Generated by Django 2.2.23 on 2021-05-26 20:05
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (bmedx) [inline ecommerce/extensions/refund/migrations/0008_auto_20210526_2005.py:6]: From the upgrade of django_extensions (via `TimeStampedModel`) These add default sorts for `latest()` on unindexed tables that have around 2M rows. The only place I can `latest` used in Ecomm and in Oscar is in tests, but there could eventually be a performance issue here. I don't think it's any more dangerous than the way things currently are, though.

File: ecommerce/invoice/migrations/0009_auto_20210526_2005.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 2.2.23 on 2021-05-26 20:05
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (bmedx) [inline ecommerce/invoice/migrations/0009_auto_20210526_2005.py:6]: Same as above, but this table only has 11k rows.

File: ecommerce/referrals/migrations/0006_auto_20210526_2005.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 2.2.23 on 2021-05-26 20:05
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (bmedx) [inline ecommerce/referrals/migrations/0006_auto_20210526_2005.py:6]: Same as above but this table has 5.5M rows.

**Comment by bmedx** [comment]
Diff for base.txt:

analytics-python==1.2.9 -> analytics-python==1.3.1
attrs==20.3.0 -> attrs==21.2.0
babel==2.9.0 -> babel==2.9.1
backoff==1.10.0 (new)
billiard==3.6.3.0 -> billiard==3.6.4.0
bleach==3.2.2 -> bleach==3.3.0
cffi==1.14.4 -> cffi==1.14.5
configparser==5.0.1 -> configparser==5.0.2
coverage==5.3.1 -> coverage==5.5
cryptography==3.3.1 -> cryptography==3.4.7
cssutils==1.0.2 -> cssutils==2.3.0
defusedxml==0.6.0 -> defusedxml==0.7.1
django-compressor==2.4 -> django-compressor==2.4.1
django-config-models==2.1.0 -> django-config-models==2.1.1
django-cors-headers==3.6.0 -> django-cors-headers==3.7.0
django-extensions==3.1.0 -> django-extensions==3.1.3
django-simple-history==2.12.0 -> django-simple-history==3.0.0
django-treebeard==4.4 -> django-treebeard==4.5.1
django-waffle==2.0.0 -> django-waffle==2.2.0
django==2.2.17 -> django==2.2.23
djangorestframework-csv==2.1.0 -> djangorestframework-csv==2.1.1
djangorestframework-datatables==0.5.2 -> djangorestframework-datatables==0.6.0
djangorestframework==3.12.2 -> djangorestframework==3.12.4
drf-extensions==0.6.0 -> drf-extensions==0.7.0
drf-jwt==1.17.3 -> drf-jwt==1.19.1
edx-auth-backends==3.3.0 -> edx-auth-backends==3.3.3
edx-django-utils==3.13.0 -> edx-django-utils==4.0.0
edx-drf-extensions==6.4.0 -> edx-drf-extensions==6.5.0
edx-opaque-keys==2.1.1 -> edx-opaque-keys==2.2.1
edx-rbac==1.3.4 -> edx-rbac==1.4.2
edx-rest-api-client==5.2.1 -> edx-rest-api-client==5.3.0
faker==5.6.5 -> faker==8.2.1
jinja2==2.11.2 -> jinja2==3.0.1
lxml==4.6.2 -> lxml==4.6.3
markupsafe==1.1.1 -> markupsafe==2.0.1
monotonic==1.6 (new)
newrelic==6.0.0.154 -> newrelic==6.4.0.157
packaging==20.8 -> packaging==20.9
pbr==5.5.1 -> pbr==5.6.0
phonenumbers==8.12.16 -> phonenumbers==8.12.23
pillow==8.1.2 -> pillow==8.2.0
purl==1.5 -> purl==1.6
pycryptodome==3.10.1 -> pycryptodome==3.9.9
pycryptodomex==3.10.1 -> pycryptodomex==3.9.9
pygments==2.7.4 -> pygments==2.9.0
pymongo==3.11.2 -> pymongo==3.11.4
rsa==4.7 -> rsa==4.7.2
rules==2.2 -> rules==3.0
six==1.15.0 -> six==1.16.0
urllib3==1.26.2 -> urllib3==1.26.5
zope.interface==5.2.0 -> zope.interface==5.4.0

**Comment by bmedx** [comment]
Diff for dev.txt:

analytics-python==1.2.9 -> analytics-python==1.3.1
attrs==20.3.0 -> attrs==21.2.0
babel==2.9.0 -> babel==2.9.1
backoff==1.10.0 (new)
billiard==3.6.3.0 -> billiard==3.6.4.0
bleach==3.2.2 -> bleach==3.3.0
cffi==1.14.4 -> cffi==1.14.5
configparser==5.0.1 -> configparser==5.0.2
coverage==5.3.1 -> coverage[toml]==5.5
cryptography==3.3.1 -> cryptography==3.4.7
cssutils==1.0.2 -> cssutils==2.3.0
ddt==1.4.1 -> ddt==1.4.2
defusedxml==0.6.0 -> defusedxml==0.7.1
diff-cover==4.2.0 -> diff-cover==5.1.1
django-compressor==2.4 -> django-compressor==2.4.1
django-config-models==2.1.0 -> django-config-models==2.1.1
django-cors-headers==3.6.0 -> django-cors-headers==3.7.0
django-extensions==3.1.0 -> django-extensions==3.1.3
django-simple-history==2.12.0 -> django-simple-history==3.0.0
django-treebeard==4.4 -> django-treebeard==4.5.1
django-waffle==2.0.0 -> django-waffle==2.2.0
django==2.2.17 -> django==2.2.23
djangorestframework-csv==2.1.0 -> djangorestframework-csv==2.1.1
djangorestframework-datatables==0.5.2 -> djangorestframework-datatables==0.6.0
djangorestframework==3.12.2 -> djangorestframework==3.12.4
docutils==0.16 -> docutils==0.17.1
drf-extensions==0.6.0 -> drf-extensions==0.7.0
drf-jwt==1.17.3 -> drf-jwt==1.19.1
edx-auth-backends==3.3.0 -> edx-auth-backends==3.3.3
edx-django-utils==3.13.0 -> edx-django-utils==4.0.0
edx-drf-extensions==6.4.0 -> edx-drf-extensions==6.5.0
edx-opaque-keys==2.1.1 -> edx-opaque-keys==2.2.1
edx-rbac==1.3.4 -> edx-rbac==1.4.2
edx-rest-api-client==5.2.1 -> edx-rest-api-client==5.3.0
faker==5.6.5 -> faker==8.2.1
gitdb==4.0.5 -> gitdb==4.0.7
gitpython==3.1.12 -> gitpython==3.1.17
importlib-metadata==4.2.0 (new)
inflect==5.0.2 -> inflect==5.3.0
jinja2==2.11.2 -> jinja2==3.0.1
lxml==4.6.2 -> lxml==4.6.3
markupsafe==1.1.1 -> markupsafe==2.0.1
monotonic==1.6 (new)
more-itertools==8.6.0 -> more-itertools==8.8.0
newrelic==6.0.0.154 -> newrelic==6.4.0.157
packaging==20.8 -> packaging==20.9
pbr==5.5.1 -> pbr==5.6.0
phonenumbers==8.12.16 -> phonenumbers==8.12.23
pillow==8.1.2 -> pillow==8.2.0
polib==1.1.0 -> polib==1.1.1
purl==1.5 -> purl==1.6
pycodestyle==2.6.0 -> pycodestyle==2.7.0
pycryptodome==3.10.1 -> pycryptodome==3.9.9
pycryptodomex==3.10.1 -> pycryptodomex==3.9.9
pygments==2.7.4 -> pygments==2.9.0
pymongo==3.11.2 -> pymongo==3.11.4
pytest-cov==2.11.1 -> pytest-cov==2.12.0
pytest-randomly==3.5.0 -> pytest-randomly==3.8.0
python-dotenv==0.15.0 -> python-dotenv==0.17.1
responses==0.12.1 -> responses==0.13.3
rsa==4.7 -> rsa==4.7.2
rules==2.2 -> rules==3.0
six==1.15.0 -> six==1.16.0
smmap==3.0.4 -> smmap==4.0.0
soupsieve==2.1 -> soupsieve==2.2.1
urllib3==1.26.2 -> urllib3==1.26.5
waitress==1.4.4 -> waitress==2.0.0
webob==1.8.6 -> webob==1.8.7
zipp==3.4.1 (new)
zope.interface==5.2.0 ->  zope.interface==5.4.0

**Comment by bmedx** [comment]
Diff for docs.txt:

babel==2.9.0 -> babel==2.9.1
docutils==0.16 -> docutils==0.17.1
jinja2==2.11.2 -> jinja2==3.0.1
markupsafe==1.1.1 -> markupsafe==2.0.1
pygments==2.7.4 -> pygments==2.9.0
six==1.15.0 -> six==1.16.0
urllib3==1.26.2 -> urllib3==1.26.5

**Comment by bmedx** [comment]
Diff for e2e.txt:

attrs==20.3.0 -> attrs==21.2.0
cffi==1.14.4 -> cffi==1.14.5
cryptography==3.3.1 -> cryptography==3.4.7
django-waffle==2.0.0 -> django-waffle==2.2.0
django==2.2.17 -> django==2.2.23
edx-django-utils==3.13.0 -> edx-django-utils==4.0.0
edx-rest-api-client==5.2.1 -> edx-rest-api-client==5.3.0
importlib-metadata==4.2.0 (new)
more-itertools==8.6.0 -> more-itertools==8.8.0
newrelic==6.0.0.154 -> newrelic==6.4.0.157
packaging==20.8 -> packaging==20.9
pbr==5.5.1 -> pbr==5.6.0
pytest-randomly==3.5.0 -> pytest-randomly==3.8.0
python-dotenv==0.15.0 -> python-dotenv==0.17.1
six==1.15.0 -> six==1.16.0
urllib3==1.26.2 -> urllib3==1.26.5
zipp==3.4.1 (new)

**Comment by bmedx** [comment]
Diff for pip_tools.txt:

click==7.1.2 -> click==8.0.1

**Comment by bmedx** [comment]
Diff for production.txt:

analytics-python==1.2.9 -> analytics-python==1.3.1
attrs==20.3.0 -> attrs==21.2.0
babel==2.9.0 -> babel==2.9.1
backoff==1.10.0 (new)
billiard==3.6.3.0 -> billiard==3.6.4.0
bleach==3.2.2 -> bleach==3.3.0
boto3==1.16.57 -> boto3==1.17.80
botocore==1.19.57 -> botocore==1.20.80
cffi==1.14.4 -> cffi==1.14.5
configparser==5.0.1 -> configparser==5.0.2
coverage==5.3.1 -> coverage==5.5
cryptography==3.3.1 -> cryptography==3.4.7
cssutils==1.0.2 -> cssutils==2.3.0
defusedxml==0.6.0 -> defusedxml==0.7.1
django-compressor==2.4 -> django-compressor==2.4.1
django-config-models==2.1.0 -> django-config-models==2.1.1
django-cors-headers==3.6.0 -> django-cors-headers==3.7.0
django-extensions==3.1.0 -> django-extensions==3.1.3
django-ses==1.0.3 -> django-ses==2.0.0
django-simple-history==2.12.0 -> django-simple-history==3.0.0
django-treebeard==4.4 -> django-treebeard==4.5.1
django-waffle==2.0.0 -> django-waffle==2.2.0
django==2.2.17 -> django==2.2.23
djangorestframework-csv==2.1.0 -> djangorestframework-csv==2.1.1
djangorestframework-datatables==0.5.2 -> djangorestframework-datatables==0.6.0
djangorestframework==3.12.2 -> djangorestframework==3.12.4
drf-extensions==0.6.0 -> drf-extensions==0.7.0
drf-jwt==1.17.3 -> drf-jwt==1.19.1
edx-auth-backends==3.3.0 -> edx-auth-backends==3.3.3
edx-django-utils==3.13.0 -> edx-django-utils==4.0.0
edx-drf-extensions==6.4.0 -> edx-drf-extensions==6.5.0
edx-opaque-keys==2.1.1 -> edx-opaque-keys==2.2.1
edx-rbac==1.3.4 -> edx-rbac==1.4.2
edx-rest-api-client==5.2.1 -> edx-rest-api-client==5.3.0
faker==5.6.5 -> faker==8.2.1
jinja2==2.11.2 -> jinja2==3.0.1
lxml==4.6.2 -> lxml==4.6.3
markupsafe==1.1.1 -> markupsafe==2.0.1
monotonic==1.6 (new)
packaging==20.8 -> packaging==20.9
pbr==5.5.1 -> pbr==5.6.0
phonenumbers==8.12.16 -> phonenumbers==8.12.23
pillow==8.1.2 -> pillow==8.2.0
purl==1.5 -> purl==1.6
pycryptodome==3.10.1 -> pycryptodome==3.9.9
pycryptodomex==3.10.1 -> pycryptodomex==3.9.9
pygments==2.7.4 -> pygments==2.9.0
pymongo==3.11.2 -> pymongo==3.11.4
rsa==4.7 -> rsa==4.7.2
rules==2.2 -> rules==3.0
s3transfer==0.3.4 -> s3transfer==0.4.2
six==1.15.0 -> six==1.16.0
urllib3==1.26.2 -> urllib3==1.26.5
zope.interface==5.2.0 -> zope.interface==5.4.0

**Comment by bmedx** [comment]
Diff for test.txt:

analytics-python==1.2.9 -> analytics-python==1.3.1
attrs==20.3.0 -> attrs==21.2.0
babel==2.9.0 -> babel==2.9.1
backoff==1.10.0 (new)
billiard==3.6.3.0 -> billiard==3.6.4.0
bleach==3.2.2 -> bleach==3.3.0
cffi==1.14.4 -> cffi==1.14.5
configparser==5.0.1 -> configparser==5.0.2
coverage==5.3.1 -> coverage[toml]==5.5
cryptography==3.3.1 -> cryptography==3.4.7
cssutils==1.0.2 -> cssutils==2.3.0
ddt==1.4.1 -> ddt==1.4.2
defusedxml==0.6.0 -> defusedxml==0.7.1
diff-cover==4.2.0 -> diff-cover==5.1.1
django-compressor==2.4 -> django-compressor==2.4.1
django-config-models==2.1.0 -> django-config-models==2.1.1
django-cors-headers==3.6.0 -> django-cors-headers==3.7.0
django-extensions==3.1.0 -> django-extensions==3.1.3
django-simple-history==2.12.0 -> django-simple-history==3.0.0
django-treebeard==4.4 -> django-treebeard==4.5.1
django-waffle==2.0.0 -> django-waffle==2.2.0
djangorestframework-csv==2.1.0 -> djangorestframework-csv==2.1.1
djangorestframework-datatables==0.5.2 -> djangorestframework-datatables==0.6.0
djangorestframework==3.12.2 -> djangorestframework==3.12.4
drf-extensions==0.6.0 -> drf-extensions==0.7.0
drf-jwt==1.17.3 -> drf-jwt==1.19.1
edx-auth-backends==3.3.0 -> edx-auth-backends==3.3.3
edx-django-utils==3.13.0 -> edx-django-utils==4.0.0
edx-drf-extensions==6.4.0 -> edx-drf-extensions==6.5.0
edx-opaque-keys==2.1.1 -> edx-opaque-keys==2.2.1
edx-rbac==1.3.4 -> edx-rbac==1.4.2
edx-rest-api-client==5.2.1 -> edx-rest-api-client==5.3.0
faker==5.6.5 -> faker==8.2.1
importlib-metadata==4.2.0 (new)
inflect==5.0.2 -> inflect==5.3.0
jinja2==2.11.2 -> jinja2==3.0.1
lxml==4.6.2 -> lxml==4.6.3
markupsafe==1.1.1 -> markupsafe==2.0.1
monotonic==1.6 (new)
more-itertools==8.6.0 -> more-itertools==8.8.0
newrelic==6.0.0.154 -> newrelic==6.4.0.157
packaging==20.8 -> packaging==20.9
pbr==5.5.1 -> pbr==5.6.0
phonenumbers==8.12.16 -> phonenumbers==8.12.23
pillow==8.1.2 -> pillow==8.2.0
polib==1.1.0 -> polib==1.1.1
purl==1.5 -> purl==1.6
pycodestyle==2.6.0 -> pycodestyle==2.7.0
pycryptodome==3.10.1 -> pycryptodome==3.9.9
pycryptodomex==3.10.1 -> pycryptodomex==3.9.9
pygments==2.7.4 -> pygments==2.9.0
pymongo==3.11.2 -> pymongo==3.11.4
pytest-cov==2.11.1 -> pytest-cov==2.12.0
pytest-randomly==3.5.0 -> pytest-randomly==3.8.0
python-dotenv==0.15.0 -> python-dotenv==0.17.1
responses==0.12.1 -> responses==0.13.3
rsa==4.7 -> rsa==4.7.2
rules==2.2 -> rules==3.0
six==1.15.0 -> six==1.16.0
soupsieve==2.1 -> soupsieve==2.2.1
urllib3==1.26.2 -> urllib3==1.26.5
waitress==1.4.4 -> waitress==2.0.0
webob==1.8.6 -> webob==1.8.7
zipp==3.4.1 (new)
zope.interface==5.2.0 -> zope.interface==5.4.0

**Comment by bmedx** [comment]
Diff for tox.txt:

packaging==20.8 -> packaging==20.9
six==1.15.0 -> six==1.16.0

**Review by inventhouse — APPROVED** [review]

**Pull request merged**